server/aws-middleware/file-api.js

```diff
@@ -0,0 +1,72 @@
+const mongoose = require('mongoose');
+const taskApi = require('../db-middleware/task-api');
+const bucket = require('../initialization/aws');
+
+const awsModule = {};
+
+const BUCKET_NAME = 'example-bucket-12345/';
+
+const uploadInput = function (file, taskId, number) {
+  return awsModule.upload({
+    Bucket: `${BUCKET_NAME}Task_${taskId}/Inputs`,
+    Key: `input${number}.txt`,
+    Body: file.data,
+  });
+};
+
+const uploadOutput = function (file, taskId, number) {
+  return awsModule.upload({
+    Bucket: `${BUCKET_NAME}Task_${taskId}/Outputs`,
+    Key: `output${number}.txt`,
+    Body: file.data,
+  });
+};
+
+awsModule.upload = function (params) {
+  bucket.createBucket(() => {
+    bucket.upload(params, (err) => {
+      if (err) {
+        console.log(err);
+      }
+    });
+  });
+};
+
+awsModule.createTask = function (file, input, length) {
+  const idBucket = new mongoose.Types.ObjectId();
+  const idInputs = [];
+  const idOutputs = [];
+
+  for (let i = 0; i < length;) {
+    uploadInput(file['input' + ++i], idBucket, i);

```

Comment:
I do not understand how it works. Did you forget that it is async code?

Looks like even if `uploadInput` fails, you'll try to `taskApi.addFile` and `taskApi.addTask`. You do not wait until any request finishing. WTF

Maybe I'm wrong, so just for misunderstanding exclusion: 
`taskApi.addFile`, `awsModule.upload`, `taskApi.addTask` are these methods async?